lib/game_logic.rb

```diff
@@ -0,0 +1,94 @@
+class Checkresult
+  attr_reader :result1
+
+  def initialize(result)
+    @result1 = result
+  end
+
+  def check_result(player, new_board)
+    i = 0
+    while i < 7
+      if new_board.array[i] != '-' && new_board.array[i] == new_board.array[i + 1] && \
+         new_board.array[i + 1] == new_board.array[i + 2] && new_board.array[i + 2] == new_board.array[i]
+        @result1 = true
+        return player
+      end
```

Comment:
This is the `if` logic I was talking about above. You see, all the blow `check_` methods (except `check_draw`) have similar `if` statements. The only difference is that they are checking different positions. So what you need to do abstract the similarities and feed the different positions to get the desired result.